src/posh-git.psm1

```diff
@@ -92,34 +89,22 @@ $GitPromptScriptBlock = {
         $promptSuffix.Text = $promptSuffix.Text.Substring(0, $promptSuffix.Text.Length - 1)
     }
 
-    # Update the host's WindowTitle is host supports it and user has not disabled $GitPromptSettings.WindowTitle
+    # Update the host's WindowTitle if host supports it and user has not disabled $GitPromptSettings.WindowTitle
     # This has to be *after* the call to Write-VcsStatus, which populates $global:GitStatus
-    if ($WindowTitleSupported) {
-        $windowTitle = $settings.WindowTitle
-        if (!$windowTitle) {
-            if ($global:PreviousWindowTitle) {
-                $Host.UI.RawUI.WindowTitle = $global:PreviousWindowTitle
+    if ($HostSupportsSettingWindowTitle -and $settings.WindowTitle) {
```

Comment:
While we're at it, should this whole block move into a `Set-WindowTitle` function?

Reply:
Sure, but we'd need to export it from the module to make it callable from the prompt func.  How about `Reset-WindowTitle` - does it need to be exported or would we be the only caller in the OnRemove event handler.